impl/client/src/test/javascript/util/xss.spec.js

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright 2010 - 2024 Hitachi Vantara.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+ define("common-ui/util/xss",  function(XSS) {
+
+         describe("sanitizeHtml", function() {
+
+             // Test for sanitizing unsafe HTML
+             it("should return correctly formatted locale", function() {
+                 const unsafeHtml = '<div>Hello</div><script>alert("xss")</script>';
+                 const expectedSanitizedHtml = '<div>Hello</div>'; // The <script> tag should be removed
+                 const result = XSS.sanitizeHtml(unsafeHtml);
+                 expect(result).toBe(expectedSanitizedHtml);
+               });
+
+              // Test for safe HTML (should remain unchanged)
+              it('should return safe HTML unchanged', () => {
+                const safeHtml = '<div><b>Hello</b></div>';
+                const result = XSS.sanitizeHtml(safeHtml);
+                expect(result).toBe(safeHtml);
+              });
+
+             // Test for empty input
+              it('should return an empty string when input is empty', () => {
+                const result = XSS.sanitizeHtml('');
+                expect(result).toBe('');
+              });
+
+              // Test for null input
+              it('should return an empty string when input is null', () => {
+                 const result = XSS.sanitizeHtml(null);
+                 expect(result).toBe('');
+              });
+
+              // Test for complex nested unsafe HTML
+              it('should remove nested unsafe elements', () => {
+                  const unsafeHtml = '<div>Hello<script>alert("xss")</script><p>World</p></div>';
+                  const expectedSanitizedHtml = '<div>Hello<p>World</p></div>'; // Remove <script>
+                  const result = XSS.sanitizeHtml(unsafeHtml);
+                  expect(result).toBe(expectedSanitizedHtml);
+              });
+
+         });
+
+
+        it("should correctly encode HTML entities", function() {
+            var map = {
+              "&": "&amp;",
+              "<": "&lt;",
+              ">": "&gt;"
+            };
+            var htmlEncoder = XSS.compileHtmlEncoder(map);
```

Comment:
`compileHtmlEncoder` is not being "exported" in `XSS`. Nor should it, imo, it's a private implementation detail. Instead, we should be testing `encodeTextForHtml` and `encodeTextForHtmlAttribute`.